src/screens/Home/Home.js

```diff
@@ -1,8 +1,23 @@
-import React from 'react'
-import Products from './components/Products'
+import React from 'react';
+import Products from './components/Products';
+import { createStackNavigator } from '@react-navigation/stack';
+import Product from './components/Product';
+import WishList from './modals/WishList';
 
-export default function Home() {
+const Stack = createStackNavigator();
+
+export default function Home({ toggleSettingsModalVisible }) {
+  function getProductOptions({ route }) {
+    return {
+      headerShown: true,
+      headerTitle: route.params.title
+    }
+  }
   return (
-    <Products />
+    <Stack.Navigator screenOptions={{ headerShown: false }}>
+      <Stack.Screen name="Products" children={() => <Products toggleSettingsModalVisible={toggleSettingsModalVisible} />} />
+      <Stack.Screen name="Product" children={(navigation) => <Product navigation={navigation} />} options={getProductOptions} />
```

Comment:
навігація і так вже передається в пропсах, не треба додатково прокидувати

```suggestion
      <Stack.Screen name="Product" component ={Product} options={getProductOptions} />
```

Reply:
Это исправил